src/Nethermind/Nethermind.JsonRpc.Test/JsonRpcSocketsClientTests.cs

```diff
@@ -146,6 +150,79 @@ static async Task<int> CountNumberOfMessages(Socket socket, CancellationToken to
             Assert.That(received, Is.EqualTo(sent));
         }
 
+        [TestCase(1)]
+        [TestCase(5)]
+        public async Task CanHandleMessageConcurrently(int concurrencyLevel)
+        {
+            CancellationTokenSource cts = new(TimeSpan.FromSeconds(10));
+            using TempPath tmpPath = TempPath.GetTempFile();
+
+            var endPoint = new UnixDomainSocketEndPoint(tmpPath.Path);
+            Socket socketListener = new(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            socketListener.Bind(endPoint);
+            socketListener.Listen(0);
+
+            using Socket sendSocket = new(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            await sendSocket.ConnectAsync(endPoint);
+
+            using IpcSocketMessageStream sendStream = new(sendSocket);
+            IJsonRpcProcessor jsonRpcProcessor = Substitute.For<IJsonRpcProcessor>();
+            Task receiver = Task.Run(async () =>
+            {
+                Socket socket = await socketListener.AcceptAsync(cts.Token);
+                using IpcSocketMessageStream stream = new(socket);
+                using JsonRpcSocketsClient<IpcSocketMessageStream> client = new(
+                    clientName: "TestClient",
+                    stream: stream,
+                    endpointType: RpcEndpoint.Ws,
+                    jsonRpcProcessor: jsonRpcProcessor,
+                    jsonRpcLocalStats: new NullJsonRpcLocalStats(),
+                    jsonSerializer: new EthereumJsonSerializer(),
+                    concurrency: concurrencyLevel
+                );
+
+                await client.ReceiveLoopAsync(cts.Token);
+            });
+
+            async Task SendTestMessage()
+            {
+                await sendStream.WriteAsync("test"u8.ToArray(), cts.Token);
+                await sendStream.WriteEndOfMessageAsync();
+            }
+
+            int concurrentCall = 0;
+            TaskCompletionSource completeSource = new TaskCompletionSource();
+            async IAsyncEnumerable<JsonRpcResult> ResponseFunc(CallInfo c)
+            {
+                Interlocked.Increment(ref concurrentCall);
+                await completeSource.Task;
+                yield return JsonRpcResult.Single(new JsonRpcSuccessResponse(null), new RpcReport());
+            }
+
+            jsonRpcProcessor
+                .ProcessAsync(Arg.Any<PipeReader>(), Arg.Any<JsonRpcContext>())
+                .Returns(ResponseFunc);
+
+            for (int i = 0; i < concurrencyLevel; i++)
```

Comment:
I'm not sure that I understand what we're testing here: we are sending `concurrencyLevel` messages sequentially, but we don't know what we're not checking what we're getting back.

Reply:
We don't care about the response, just that we can have `concurrencyLevel` of in flight processing.